js/main.js

```diff
@@ -50,18 +57,84 @@ function isWeaponFilteredOut(weaponName, isOffhand) {
         case 'Small Shield':
         case 'Medium Shield':
         case 'Greatshield':
+        case 'Thrusting Shield':
             return FILTER_NO_MAINHAND_SHIELD.checked;
         case 'Glintstone Staff':
             return FILTER_NO_MAINHAND_STAFF.checked;
         case 'Sacred Seal':
             return FILTER_NO_MAINHAND_SEAL.checked;
+        case 'Light Bow': 
+        case 'Bow':
+        case 'Greatbow':
         case 'Crossbow':
-            return FILTER_NO_MAINHAND_CROSSBOW.checked;
+            return FILTER_NO_MAINHAND_BOW.checked;
+        case 'Torch':
+            return FILTER_NO_MAINHAND_TORCH.checked;
+        case 'Perfume Bottle':
+            return FILTER_NO_MAINHAND_PERFUME_BOTTLE.checked;
+    }
+    return false;
+}
+function enableCheckboxes(checkboxId, isEnabled){
```

Comment:
Please respect the original brace style, which is to put a space between the closing parenthesis of the parameter list and the opening brace.